src/v2/models/application.js

```diff
@@ -878,6 +878,33 @@ export default class ApplicationModel extends GenericModel {
     return _.filter(namespaces, (ns) => !_.get(ns, 'metadata.name', '').startsWith('openshift') && !_.get(ns, 'metadata.name', '').startsWith('open-cluster-management'));
   }
 
+  async getManagedCluster(clusterName) {
+    const managedCluster = await this.kubeConnector.get(
+      `/apis/cluster.open-cluster-management.io/v1/managedclusters/${clusterName}`,
+    ).catch((err) => {
+      logger.error(err);
+      throw err;
+    });
+    let successImportStatus = true;
+    if (managedCluster.status.conditions) {
+      // check cluster import condition
+      const managedClusterCondition = _.get(managedCluster, 'status.conditions');
+      _.forEach(managedClusterCondition, (value) => {
+        if (_.has(value, 'status')) {
+          const status = _.get(value, 'status');
+          if (status !== 'True') {
+            successImportStatus = false;
+          }
+        }
+      });
```

Comment:
Might be better to check only `ManagedClusterConditionAvailable`, which I believe implies that the cluster is accepted and joined. This would be more future-proof, because consider if a new condition were added at some point that indicates something bad about the cluster. We wouldn't want to require that bad condition to be True.